conf/templates/README.md

```diff
@@ -0,0 +1,188 @@
+## Muchos EC2 Cluster Templates
+
+Cluster templates are intended to provide greater flexibility, if needed,
+with respect to instance type selection and launch configuration for
+your EC2 hosts. For example, cluster templates may be ideal for use
+cases that require distinct, per-host launch configurations, and
+for use cases that require hosts to have persistent, EBS-backed data
+volumes (rather than ephemeral volumes, the muchos default for EC2
+clusters)
+
+If you are already familiar with muchos and with the basics of EC2
+launch requests, then creating your own launch templates will be simple
+and straightforward.
+
+Please follow the guidance provided here to ensure compatibility between
+your custom templates and muchos automation
+
+## Configuration
+
+### Select a cluster template in *muchos.props*
+```
+[ec2]
+...
+cluster_template = example
+...
+```
+The configured value must match the name of a subdirectory under
+`conf/templates`
+
+```
+~$ ls -1a fluo-muchos/conf/templates/example
+.
+..
+devices
+metrics.json
+namenode.json
+resourcemanager.json
+zookeeper.json
+worker.json
+```
+The subdirectory will contain one or more user-defined EC2 launch
+templates (*\*.json*) for your various host types, and it will
+include a *devices* file specifying the desired mount points for all
+data volumes (excluding root volumes, as they are mounted
+automatically)
+
+### Defining EC2 launch templates and device mounts for your hosts
+
+#### Launch Templates: *{service-name}.json* files
+
+Each JSON file represents a standard EC2 launch request, and each file
+name must match one of the predefined muchos service names, as
+defined in the **nodes** section of *muchos.props*. E.g.,
+```
+... 
+[nodes]
+leader1 = namenode,resourcemanager,accumulomaster
+leader2 = metrics,zookeeper
+worker1 = worker
+worker2 = worker
+worker3 = worker
+worker4 = worker
+```
+In template mode, the first service listed for a given host denotes the
+template to be selected for its launch.
+
+Based on the example given above:
+* **leader1** selects **namenode.json**
```

Comment:
`leader1` also list a `resourcemanager` in the example, so what happens if `namenode.json` and `resourcemanager.json ` exists?

Reply:
> `leader1` also list a `resourcemanager` in the example, so what happens if `namenode.json` and `resourcemanager.json ` exists?

In that case, `resourcemanager.json` is simply ignored, since `namenode` is the first service listed for that host. Good point though...I realize now that the example I provided should have omitted all but `namenode.json`, `metrics.json`, and `worker.json`, since those are the only relevant files, based on the given the `nodes` config

In retrospect, perhaps it would be more intuitive to have **{hostname}.json** files instead, at for non-worker node types, and then for the worker template (which presumably would be common to _all_ worker nodes) either require "worker.json" to exist, or have the user explicitly identify the "worker_template_name" in muchos.props

Reply:
Perhaps instead, we can have some configuration syntax that uses an explicit **{templatename}**?

Something like:
```ini
[nodes]
leader1 = namenode,resourcemanager;template={name}
```

If unspecified, the template name could be `default` or `worker` or whatever makes things easiest.

Reply:
> 
> Something like:
> 
> ```ini
> [nodes]
> leader1 = namenode,resourcemanager/{templatename}
> ```
> 
> If unspecified, the template name could be `default` or `worker` or whatever makes things easiest.

That sounds good to me

Reply:
If the EC2-python refactor moves ahead (and "default mode" functionality is removed), then I also foresee a need to rework the _devices_ file design and related Ansible scripts, due to the limitations it imposes on your non-worker (default) nodes. That is, you can only leverage distinct instance types if those types all represent device names in the exact same way. E.g., you can't have some with **/dev/xvdf** and others with **/dev/nvme\***, etc

Reply:
```
[nodes]
leader1 = namenode,resourcemanager/{templatename}
```

Since templates are an EC2 specific feature and currently `[nodes]` is not outside of the EC2 section of the config, its worth considering how that change would impact the usability of Azure and Bare Metal clusters.